File: code/B_14888.py
from itertools import permutations

N=int(input())

num = list(map(int,input().split(" ")))
oper=list(map(int,input().split(" ")))

oper_model=[]
for idx,val in enumerate(oper):
    for j in range(val):
        if(idx==0):
            oper_model.append("+")
        elif(idx==1):
            oper_model.append("-")
        elif(idx==2):
            oper_model.append("*")
        elif(idx==3):   
            oper_model.append("/")

oper_p = list(permutations(oper_model,N-1))            


max_v=-1000000000
min_v=1000000000

for val1 in oper_p:
    t1=num[0]
    for idx,val in enumerate(val1):
        t2=num[idx+1]
        if(val == '+'):
            t1 = t1+t2
        elif(val=='-'):
            t1 = t1 - t2
        elif(val == '/'):
            if(t1<0):
                t1 = -(abs(t1)//t2)
            else:
                t1 = t1//t2    
        elif(val =='*'):
            t1 = t1*t2

    max_v = max(max_v,t1)
    min_v = min(min_v,t1)

print(max_v)
print(min_v)    

File: code/B_14888.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int n;
vector<int> num;
vector<char> oper;
vector<int> oper_n;
int min_v=1000000000;
int max_v=-1000000000;


int main() {
	ios::sync_with_stdio(false);
	
	cin >> n;
	for (int i = 0; i < n; i++)
	{
		int temp;
		cin >> temp;
		num.push_back(temp);
	}
	int cnt = 0;
	for (int i = 0; i < 4; i++)
	{
		int temp;
		cin >> temp;;
		for (int j = 0; j < temp; j++)
		{
			if (i == 0)
				oper.push_back('+');
			else if(i==1)
				oper.push_back('-');
			else if (i == 2)
				oper.push_back('*');
			else if (i == 3)
				oper.push_back('/');
			oper_n.push_back(cnt++);
		}
	}

	int t1 = num[0];
	int t2;
	for (int i = 0; i < oper.size(); i++)
	{
		t2 = num[i + 1];
		if (oper[i] == '+')
			t1 = t1 + t2;
		else if (oper[i] == '-')
			t1 = t1 - t2;
		else if (oper[i] == '*')
			t1 = t1 * t2;
		else if (oper[i] == '/')
			t1 = t1 / t2;
	}

	max_v = max(max_v, t1);
	min_v = min(min_v, t1);


	
	while (next_permutation(oper_n.begin(), oper_n.end()))
	{
		t1 = num[0];
		for (int i = 0; i < oper.size(); i++)
		{
			t2 = num[i + 1];
			if (oper[oper_n[i]] == '+')
				t1 = t1 + t2;
			else if (oper[oper_n[i]] == '-')
				t1 = t1 - t2;
			else if (oper[oper_n[i]] == '*')
				t1 = t1 * t2;
			else if (oper[oper_n[i]] == '/')
				t1 = t1 / t2;
		}
		max_v = max(max_v,t1);
		min_v = min(min_v,t1);
	}

	cout << max_v << "\n" << min_v << "\n";

	return 0;
}